sklearn/metrics/tests/test_ranking.py

```diff
@@ -371,6 +371,20 @@ def test_roc_curve_drop_intermediate():
                               [1.0, 0.9, 0.7, 0.6, 0.])
 
 
+def test_roc_curve_fpr_tpr_increasing():
```

Comment:
I feel like the fact that elements are sorted for one random sample isn't a very strong assurance. There are edge cases that could be further tested (such as having repeated thresholds), too, but I'm not sure what reasonable edge cases for this test are.

Reply:
Basically the edge cases are when the definition of fps are not equal because of floating point errors:
```py
tps = stable_cumsum(y_true * weight)[threshold_idxs]
fps = stable_cumsum(weight)[threshold_idxs] - tps
fps = stable_cumsum((1 - y_true) * weight)[threshold_idxs]
```

It is not obvious to me how to construct simply an example that does not work but maybe with a little bit of thought there is a way to put a simpler one together.

For full details the best is to look at the definition of _binary_clf_curve, especially how the other variables are defined.

Reply:
OK I found a simpler example:

```py
def test_roc_curve_fpr_tpr_increasing():
    # Ensure that fpr and tpr returned by roc_curve are increasing
    # Construct an edge case with float y_score and sample_weight
    # when some adjacent values of fpr and tpr are the same.
    y_true = [0, 0, 1, 1, 1]
    y_score = [0.1, 0., 0.3, 0.4, 0.5]
    sample_weight = np.repeat(0.2, 5)
    fpr, tpr, _ = roc_curve(y_true, y_score,
                            sample_weight=sample_weight)
    assert_equal((np.diff(fpr) < 0).sum(), 0)
    assert_equal((np.diff(tpr) < 0).sum(), 0)
```

Are you happier with this one @jnothman?

Reply:
```
(Pdb) np.diff(fpr)
array([  5.00000000e-01,   0.00000000e+00,   2.22044605e-16,
        -3.33066907e-16,   5.00000000e-01])
```